packages/runtime/datastore/src/channelStorageService.ts

```diff
@@ -50,7 +50,16 @@ export class ChannelStorageService implements IChannelStorageService {
             ? (await this.extraBlobs).get(id)
             : undefined;
 
-        return blob ?? this.storage.read(id);
+        return blob ?? this.storage.readString(id);
+    }
+
+    public async readString(path: string): Promise<string> {
+        const id = await this.getIdForPath(path);
+        const blob = this.extraBlobs !== undefined
+            ? (await this.extraBlobs).get(id)
```

Comment:
What data format is this.extraBlobs? If it's base64 encoded data, than this code needs to decode utf-8 string out of it.

Reply:
From existing code, this.extraBlobs is base64-encoded data. So this code needs to decode it.
I'd open an issue for some future to re-evaluate it, i.e. we are spending cycles to encode strings into base64 and then back. If we change the users of this class to provide us binary data instead (not strings, but buffers), then we can avoid this useless conversion back and forth. But this can be done as separate item